REGRESSION(274824@main): [MSE][GStreamer][Debug] SHOULD NEVER BE REACHED in MediaPlayerPrivateGStreamerMSE::buffered()

obyknovenius opened the pull request:
#### df3b091edfbab831015cdce7961af8b1c2a93c83
<pre>
REGRESSION(274824@main): [MSE][GStreamer][Debug] SHOULD NEVER BE REACHED in MediaPlayerPrivateGStreamerMSE::buffered()
<a href="https://bugs.webkit.org/show_bug.cgi?id=269988">https://bugs.webkit.org/show_bug.cgi?id=269988</a>

Reviewed by Philippe Normand.

`MediaPlayerPrivateGStreamerMSE` should overload `extraMemoryCost()` and
return 0, as `MediaPlayerPrivateMediaSourceAVFObjC` already does.

* Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp:
(WebCore::MediaPlayerPrivateGStreamerMSE::extraMemoryCost const):
* Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.h:

Canonical link: <a href="https://commits.webkit.org/275317@main">https://commits.webkit.org/275317@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2576d1717ed64a8e9d3813fcc2c3bb5123973cac

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/41512 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/20526 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/43890 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/44080 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/37604 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/43819 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/23632 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/17856 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/34326 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/42086 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/17453 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/35753 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/14987 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/15169 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/36766 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/45442 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/37700 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/37078 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/40835 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/16327 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/13414 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/39243 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/17946 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9304 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/18001 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/17590 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

jyavenard reviewed (changes requested):
this feels very wrong.

The memory used by a MediaSource shouldn't be accounted for the memory of a HTMLMediaElement.
The buffered range is also irrelevant to the memory usage of the HTMLMediaElement when it comes to MSE. The memory size is already accounted by each SourceBuffer.

If it was used by the MediaPlayerPrivateGStreamerMSE, the fix for this issue is to overload extramMemoryCost() like MediaPlayerPrivateMediaSourceAVFObjC or MediaPlayerPlayerMediaStreamAVFObjC and return 0.

In addition using this totalDuration / duration() is a nonsensical value that doesn't represent the memory usage of anything.

It should be ditched IMHO, but that's a different issue altogether

obyknovenius committed:
REGRESSION(274824@main): [MSE][GStreamer][Debug] SHOULD NEVER BE REACHED in MediaPlayerPrivateGStreamerMSE::buffered()
https://bugs.webkit.org/show_bug.cgi?id=269988

Reviewed by Philippe Normand.

`MediaPlayerPrivateGStreamerMSE` should overload `extraMemoryCost()` and
return 0, as `MediaPlayerPrivateMediaSourceAVFObjC` already does.

* Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp:
(WebCore::MediaPlayerPrivateGStreamerMSE::extraMemoryCost const):
* Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.h:

Canonical link: https://commits.webkit.org/275317@main
